code/modules/mining/machine_redemption.dm

```diff
@@ -160,11 +174,49 @@
 	else
 		unregister_input_turf() // someone just un-wrenched us, unregister the turf
 
+/obj/machinery/mineral/ore_redemption/can_be_unfasten_wrench(mob/user, silent)
+	if(welded)
+		if(!silent)
+			to_chat(user, span_warning("[src] is welded to the floor!"))
+		return FAILED_UNFASTEN
+
+	return ..()
+	
 /obj/machinery/mineral/ore_redemption/wrench_act(mob/living/user, obj/item/tool)
 	. = ..()
 	default_unfasten_wrench(user, tool)
 	return TOOL_ACT_TOOLTYPE_SUCCESS
+	
+/obj/machinery/mineral/ore_redemption/welder_act(mob/living/user, obj/item/item)
+	..()
+	if(welded)
+		if(!item.tool_start_check(user, amount=0))
+			return TRUE
+		user.visible_message(span_notice("[user.name] starts to cut the [name] free from the floor."), \
+			span_notice("You start to cut [src] free from the floor..."), \
+			span_hear("You hear welding."))
+		if(!item.use_tool(src, user, 20, 1, 50))
+			return FALSE
+		welded = FALSE
+		to_chat(user, span_notice("You cut [src] free from the floor."))
+		update_cable_icons_on_turf(get_turf(src))
+		return TRUE
 
+	if(!anchored)
+		to_chat(user, span_warning("[src] needs to be wrenched to the floor!"))
+		return TRUE
+	if(!item.tool_start_check(user, amount=0))
+		return TRUE
+	user.visible_message(span_notice("[user.name] starts to weld the [name] to the floor."), \
+		span_notice("You start to weld [src] to the floor..."), \
+		span_hear("You hear welding."))
+	if(!item.use_tool(src, user, 20, 1, 50))
+		return FALSE
+	welded = TRUE
+	to_chat(user, span_notice("You weld [src] to the floor."))
+	update_cable_icons_on_turf(get_turf(src))
```

Comment:
why?

Reply:
> why?

Leftover from emitter code. Gonna make all the changes you requested as well as requiring it to be unwelded to open the panel..